app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -929,10 +930,44 @@ def private_load_allowed_networks(vlans, uid_ems_cluster)
       vlans['<Template>'] = _('<Use template nics>')
     end
 
+    def private_load_allowed_networks(vlans, uid_ems_cluster)
+      profiles = local_get_vnic_profiles_in_cluster(uid_ems_cluster)
+      profiles.each do |profile, profile_network|
+        vlans[profile.uid_ems] = "#{profile.name} (#{profile_network.name})"
+      end
+
+      vlans['<Empty>'] = _('<No Profile>')
+      vlans['<Template>'] = _('<Use template nics>')
+    end
+
+    def local_get_vnic_profiles_in_cluster(uid_ems_cluster)
+      cluster_profiles = {}
+      profiles = all_vnic_profiles_as_lans_in_ems(ext_management_system)
+      cluster_networks = all_networks_as_switches_in_cluster(uid_ems_cluster)
+      profiles.each do |p|
+        profile_network = cluster_networks.detect { |n| n.id == p.switch.id }
+        if profile_network
+          cluster_profiles[p] = profile_network
+        end
+      end
+      cluster_profiles
+    end
+
+    def all_networks_as_switches_in_cluster(uid_ems_cluster)
+      cluster = EmsCluster.find_by(:uid_ems => uid_ems_cluster)
+      hosts = cluster.hosts
+      host_switches = hosts.collect(&:host_switches).flatten.compact.uniq
+      host_switches.collect(&:switch).flatten.compact.uniq
+    end
+
+    def all_vnic_profiles_as_lans_in_ems(ems)
+      ManageIQ::Providers::Redhat::InfraManager::DistributedVirtualSwitch.where(:ems_id => ems.id).collect(&:lans).flatten.compact.uniq
+    end
+
     def parse_vnic_profile_id(requested_profile, uid_ems_cluster)
       if requested_profile.include?('(')
         vlans = {}
-        private_load_allowed_networks(vlans, uid_ems_cluster)
+        private_load_allowed_networks_from_provider(vlans, uid_ems_cluster)
```

Comment:
Does this still do a direct API call to the provider?

Reply:
Yes, but this is not called from the UI, this is called during the run of the provisioning worker itself: https://github.com/ManageIQ/manageiq-providers-ovirt/blob/master/app/models/manageiq/providers/redhat/infra_manager/provision/configuration/network.rb#L22

If I understand correctly this is not a problem.

Reply:
Okay